src/ui/public/utils/scanner.js

```diff
@@ -38,6 +38,10 @@ Scanner.prototype.scanAndMap = function (searchString, options, mapFn) {
 
   return new Promise((resolve, reject) => {
     const getMoreUntilDone = (error, response) => {
+      if (error) {
```

Comment:
Can you add a test to verify that when an error occurs, the scan promise is rejected?